### PR TITLE
DI-1299 Check Dias somalier results in GRCh37 vs GRCh38

### PR DESCRIPTION
**Pull request by rklocke**
- Add Python script to find and read in `Multiqc_somalier.samples.tsv` files in GRCh37 and GRCh38 and find any samples with mismatches for `Predicted_Sex` between genome builds
- Add config files given as input to script for TWE and CEN
- Add README for script
- Update repo README for this request

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)

## Summary by CodeRabbit

- **New Features**
	- Introduced two new configuration files for assay parameters.
	- Added a new script for comparing genomic data between GRCh37 and GRCh38.
	- Updated the `README.md` with a new section detailing the comparison script and its functionality.

- **Documentation**
	- Enhanced `README.md` with details on the new script and configuration files.
	- Added a new entry for ticket [DI-1299] in the ticket summary table of `README.md`.
	- Updated the link for ticket [DI-1294] in `README.md`.

- **Bug Fixes**
	- Updated `.gitignore` to ensure specific configuration files are not ignored.

[DI-1299]: https://cuhbioinformatics.atlassian.net/browse/DI-1299?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jethror1 — CHANGES_REQUESTED** [review]
> Reviewed 4 of 6 files at r1, 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: 5 of 6 files reviewed, 28 unresolved discussions (waiting on @Addy81 and @rklocke)

___
*[`DI-1299/compare_b37_and_b38.py` line 110 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV029L9Q8mH8wvkBY1:-OEV029M2Dlr-hCT-n6D:b-hmidqr) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L110)):*
fyi you can simplify this into the return like this
```
>>> def func():
...     d = {'a': 'foo', 'b': 'bar', 'c': 'baz'}
...     keys = ['a', 'b']
...     
...     return list(map(d.get, keys))
... 
>>> 
>>> foo, bar = func()
>>> foo
'foo'
>>> bar
'bar'
```

the `map` with `get` is nice, I approve



_Code quote:_
```Python
    (
        assay,
        search_term,
        number_of_projects,
        filename,
        column_to_compare,
        sample_column,
        variables_to_plot,
    ) = list(map(config_dict.get, keys))

    return (
        assay,
        search_term,
        number_of_projects,
        filename,
        column_to_compare,
        sample_column,
        variables_to_plot,
    )
```
___
*[`DI-1299/compare_b37_and_b38.py` line 240 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV0dJN4zne9u72jHfi:-OEV0dJN4zne9u72jHfj:brcrq3i) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L240)):*
> ```Python
> 
>     if non_live_files:
>         for non_live_file in non_live_files:
> ```

not for now, but remind me to share an example of bulk unarchiving per project of files from a big list, I've a nice example somewhere

**Review by rklocke — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: 3 of 6 files reviewed, 32 unresolved discussions (waiting on @Addy81, @coderabbitai[bot], and @jethror1)

___
*[`DI-1299/compare_b37_and_b38.py` line 110 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV029L9Q8mH8wvkBY1:-OEYLRGV5_uTT6HJc23X:bcsj0jx) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L110)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

fyi you can simplify this into the return like this
```
>>> def func():
...     d = {'a': 'foo', 'b': 'bar', 'c': 'baz'}
...     keys = ['a', 'b']
...     
...     return list(map(d.get, keys))
... 
>>> 
>>> foo, bar = func()
>>> foo
'foo'
>>> bar
'bar'
```

the `map` with `get` is nice, I approve
</blockquote></details>

this is nice, have updated
___
*[`DI-1299/compare_b37_and_b38.py` line 240 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV0dJN4zne9u72jHfi:-OEYLg7CDDV5GF5unU3Q:b-896fix) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L240)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

not for now, but remind me to share an example of bulk unarchiving per project of files from a big list, I've a nice example somewhere
</blockquote></details>

Done.

File: README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Repository for code related to small tasks for supporting the RD service.
 | [DI-1189] | Create new PanelApp JSON file
 | [DI-435] | Find and merge VCFs for creation of a GRCh38 POP AF VCF
 | [DI-1294] | Remove any content from the PID fields in workbooks in a given path
+| [DI-1299] | Compare values between GRCh37 and GRCh38 for Somalier
```
[the remark this answers is not in the document]
Comment (rklocke) [reply README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -18,4 +19,5 @@
 [DI-1094]: https://cuhbioinformatics.atlassian.net/browse/DI-1094
 [DI-1189]: https://cuhbioinformatics.atlassian.net/browse/DI-1189
 [DI-435]: https://cuhbioinformatics.atlassian.net/browse/DI-435
-[DI-1294]: https://cuhbioinformatics.atlassian.net/issues/DI-1294
+[DI-1294]: https://cuhbioinformatics.atlassian.net/issues/DI-1294
```
[the remark this answers is not in the document]
Comment (rklocke) [reply README.md]: Done.

File: README.md [hunk 3]
```diff
@@ -18,4 +19,5 @@
...
+[DI-1299]: https://cuhbioinformatics.atlassian.net/browse/DI-1299
```
[the remark this answers is not in the document]
Comment (rklocke) [reply README.md]: Done.

File: DI-1299/compare_b37_and_b38.py [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+    Returns
+    -------
+    json_dict : dict
+        the JSON converted to a Python dictionary
+    """
+    with open(file_name, "r", encoding="utf8") as json_file:
+        json_dict = json.load(json_file)
+
+    return json_dict
+
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/compare_b37_and_b38.py:54]: Not necessary

File: DI-1299/compare_b37_and_b38.py [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+        )
+    )
+    assert projects, f"No projects found with the search term {search_term}"
+    projects = sorted(projects, key=lambda x: x["describe"]["name"])
+
+    if number_of_projects is not None:
+        projects = projects[-int(number_of_projects) :]
+
+    return projects
+
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/compare_b37_and_b38.py]: Done.

File: DI-1299/compare_b37_and_b38.py [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+    match.group(0) : str
+        name of the sequencing run
+    """
+    match = re.search(r"\d{6}_[A-Za-z0-9]+_\d{4}_[A-Z0-9]+", b38_project_name)
+    assert (
+        match
+    ), f"Error - no sequencing run name extracted from {b38_project_name}"
+
+    return match.group(0)
+
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/compare_b37_and_b38.py]: Done.

File: DI-1299/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        ]
+    ]
+}
+```
+
+### Running
+Example command:
+```
+python compare_b37_and_b38.py --config TWE_config.json
+```
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/README.md]: Done.

File: DI-1299/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+### How it works
+The script:
+- Find all projects using the `search_term` in GRCh38 in DNAnexus.
+- Gets the related GRCh37 project for each GRCh38 project with the prefix `002_` and the suffix `_{assay}`.
+ - Within the GRCh37 and GRCh38 projects, find all files by `filename` and reads them all in with pandas.
+- Identifies any mismatches in `column_to_compare` between the GRCh37 and GRCh38 values for each sample and investigate.
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/README.md]: This is an indented bullet, fixed

File: DI-1299/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+## How it works
+The script:
+- Find all projects using the `search_term` in GRCh38 in DNAnexus.
+- Gets the related GRCh37 project for each GRCh38 project with the prefix `002_` and the suffix `_{assay}`.
+  - Within the GRCh37 and GRCh38 projects, find all files by `filename` and reads them all in with pandas.
+- Identifies any mismatches in `column_to_compare` between the GRCh37 and GRCh38 values for each sample and investigate.
+- If mismatches are found, plots the variables in `variables_to_plot` to look at differences in these values for each sample between genome builds.
+
+## Output
```
[the remark this answers is not in the document]
Comment (rklocke) [reply DI-1299/README.md:50]: No

**Review by rklocke — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: 3 of 6 files reviewed, 32 unresolved discussions (waiting on @Addy81, @coderabbitai[bot], and @jethror1)

___
*[`DI-1299/compare_b37_and_b38.py` line 110 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV029L9Q8mH8wvkBY1:-OEYLRGV5_uTT6HJc23X:bcsj0jx) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L110)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

fyi you can simplify this into the return like this
```
>>> def func():
...     d = {'a': 'foo', 'b': 'bar', 'c': 'baz'}
...     keys = ['a', 'b']
...     
...     return list(map(d.get, keys))
... 
>>> 
>>> foo, bar = func()
>>> foo
'foo'
>>> bar
'bar'
```

the `map` with `get` is nice, I approve
</blockquote></details>

this is nice, have updated
___
*[`DI-1299/compare_b37_and_b38.py` line 240 at r2](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-OEV0dJN4zne9u72jHfi:-OEYLg7CDDV5GF5unU3Q:b-896fix) ([raw file](https://github.com/eastgenomics/RD_requests/blob/79c054fa80a37602fc402c3a1a343d23a6fff40a/DI-1299/compare_b37_and_b38.py#L240)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

not for now, but remind me to share an example of bulk unarchiving per project of files from a big list, I've a nice example somewhere
</blockquote></details>

Done.

**Review by jethror1 — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-:-OEYO5Bw9WxdZmGSNo-J:bnfp4nl)* status: 3 of 6 files reviewed, 31 unresolved discussions (waiting on @Addy81, @coderabbitai[bot], and @rklocke)

**Review by Addy81 — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 3 of 3 files at r3, all commit messages.
> Dismissed @coderabbitai[bot] from 31 discussions.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: all files reviewed, 20 unresolved discussions (waiting on @coderabbitai[bot] and @jethror1)

**Review by Addy81 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: all files reviewed, 20 unresolved discussions (waiting on @Addy81, @coderabbitai[bot], and @jethror1)

**Review by Addy81 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22#-:-OEZ1rjn5KDbn3hvRMH4:bnfp4nl)* status: all files reviewed, 20 unresolved discussions (waiting on @Addy81, @coderabbitai[bot], and @jethror1)

**Review by Addy81 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/RD_requests/22)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @rklocke)

**Pull request merged**